src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesTests.cs

```diff
@@ -36253,7 +36290,10 @@ static void F1<T1>(T1 t1)
                 Diagnostic(ErrorCode.WRN_ConvertingNullableToNonNullable, "(object)t1").WithLocation(8, 11),
                 // (8,11): warning CS8604: Possible null reference argument for parameter 'o' in 'void C.F(object o)'.
                 //         F((object)t1);
-                Diagnostic(ErrorCode.WRN_NullReferenceArgument, "(object)t1").WithArguments("o", "void C.F(object o)").WithLocation(8, 11));
+                Diagnostic(ErrorCode.WRN_NullReferenceArgument, "(object)t1").WithArguments("o", "void C.F(object o)").WithLocation(8, 11),
+                // (14,13): hidden CS8605: Result of the comparison is possibly always true.
```

Comment:
>hidden CS8605: Result of the comparison is possibly always true. [](start = 28, length = 64)

I didn't follow, why are we getting this diagnostic?

Reply:
T2 is constrained to `class!`, so `t2` should never be null.

---
In reply to: [207682296](https://github.com/dotnet/roslyn/pull/28956#discussion_r207682296) [](ancestors = 207682296)

Reply:
Thanks for the clarification.

---
In reply to: [207976300](https://github.com/dotnet/roslyn/pull/28956#discussion_r207976300) [](ancestors = 207976300,207682296)